packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -183,7 +194,31 @@ class ${class_name}:
             Stream of measurement output.
         """
         parameter_values = [${measure_api_parameters}]
-        request = self._create_measure_request(parameter_values)
-
-        for response in self._get_stub().Measure(request):
-            yield self._deserialize_response(response)
+        with self._initialization_lock:
+            if self._measure_response is not None:
+                print("A measure call is already in progress.")
+                return
+            request = self._create_measure_request(parameter_values)
+            self._measure_response = self._get_stub().Measure(request)
+
+        try:
+            for response in self._measure_response:
+                % if output_metadata:
+                yield self._deserialize_response(response)
+                % else:
+                yield
+                % endif
+        except grpc.RpcError as e:
+            if e.code() == grpc.StatusCode.CANCELLED:
+                print("The measure call is canceled.")
+            else:
+                raise
+        finally:
+            self._measure_response = None
```

Comment:
Similar to the `_get_stub` method. Can we also place this try...except block within the `initialization_lock`?

Reply:
To restrict the users to avoid calling `measure` multiple time simultaneuosly from diffrent threads, placed this try...except block outside the `initialization_lock`.